### ISSUE TITLE
Missing lib files in both new and existing projects

### ISSUE THREAD

**Issue by cooljeffrey** [maqetta/maqetta#3813]
Version : Latest GIT version
Reproduce Steps :
1. Create a new Desktop Application, or open an existing project.
2. Open some html file, the VE always shows "Loading ...". The console shows NPE.
Console Logs :

> Mar 12, 2013 1:31:30 PM org.davinci.ajaxLibrary.LibraryManager$BundleLibraryInfo getUri
> SEVERE: Library file not found! :WebContent/dojo
> Mar 12, 2013 1:31:30 PM org.davinci.ajaxLibrary.LibraryManager$BundleLibraryInfo getUri
> SEVERE: Library file not found! :WebContent/maqetta/themes

And in maqetta.core.server.user.User ::: findFiles , the following code might throw out NPE if "start == null" .

``` java
            IVResource start = null;
            if (startFolder == null || startFolder.equals(".")) {
                  start = this.workspace;
            } else {
                 start = this.getResource(startFolder);
            }

            results.addAll(Arrays.asList((start.find(pathStr))));
```

**Comment by cooljeffrey** [comment]
Checked the dojo plugin, it is up and running.

> id  State       Bundle
> 0   ACTIVE      org.eclipse.osgi_3.8.1.v20120830-144521
>                 Fragments=17
> 1   ACTIVE      org.eclipse.equinox.simpleconfigurator_1.0.301.v20120828-033635
> 2   ACTIVE      ch.qos.logback.classic_1.0.0.v20111214-2030
>                 Fragments=104
> 3   ACTIVE      ch.qos.logback.core_1.0.0.v20111214-2030
> 4   RESOLVED    ch.qos.logback.slf4j_1.0.0.v20120123-1500
>                 Master=112
> 5   ACTIVE      com.ibm.icu_4.4.2.v20110823
> 6   ACTIVE      com.jcraft.jsch_0.1.44.v201101211721
> 7   ACTIVE      com.sun.el_2.2.0.v201108011116
> 8   ACTIVE      com.worklight.worklight-3rd-parties_5.0.2.407
> 9   ACTIVE      davinci.dojo_1_8_1.0.0.qualifier
> 10  ACTIVE      davinci.gridx_1.0.0.qualifier
> 11  ACTIVE      davinci.html_1.0.0.qualifier
> 12  ACTIVE      javax.activation_1.1.0.v201108011116
> 13  ACTIVE      javax.el_2.2.0.v201108011116
> 14  ACTIVE      javax.mail.glassfish_1.4.1.v201108011116

**Comment by billreed63** [comment]
I just pulled from git an do not see this error, did you do a submodule update? Have you been running successfully from eclipse prior to this?

**Comment by cooljeffrey** [comment]
Yes it did run successfully before I made a pull, then I got this issue. To find the reason, I did cloned the repo again, with submodules, but still got this. All I changed is just removing zazl in the launch config.

**Comment by billreed63** [comment]
Have you tried a clean user workspace?

**Comment by cooljeffrey** [comment]
Yes, i did deleted all stuff inside my maqetta workspace : __maqetta_server_workspace , including all hidden files/folders, once I registered a new account and tried to create a new desktop applicaiton, I got exactly same errors. Maqetta still failed to find the dojo and themes, and still threw NPE on that method.

**Comment by cooljeffrey** [comment]
After several times of clean clone and tries, I resolved this issue. I am still not sure about what was wrong with the git or repo, however I confirmed that taking use of Maqetta nightly builds ( like 0312 ) as target runtime would cause some compile errors on some code referencing to Orion, because the dependent orion version was not matched, once I set up the target runtime by using Orion 1.0 build everything is working well now. Thanks for help!
